src/Server/passport/lib/app/lifecycle.ts

```diff
@@ -31,7 +31,38 @@ module.exports.onLocalLogin = function (
   passport.authenticate("local-with-otp")(req, res, function (err: any) {
     if (req.xhr) {
       if (err) {
-        return res.status(500).send({ success: false, error: err.message })
+        switch (true) {
+          case err.message?.includes("invalid email or password"): {
+            return res.status(403).send({ success: false, error: err.message })
+          }
+          case err.message?.includes(
+            "missing two-factor authentication code"
+          ): {
+            return res.status(403).send({ success: false, error: err.message })
+          }
+          case err.message?.includes(
+            "invalid two-factor authentication code"
+          ): {
+            return res.status(403).send({ success: false, error: err.message })
+          }
+          case err.message?.includes(
+            "account locked, try again in a few minutes"
+          ): {
+            return res.status(403).send({ success: false, error: err.message })
+          }
+          case err.message?.includes("User Already Exists"): {
+            return res.status(409).send({ success: false, error: err.message })
+          }
+          case err.message?.includes("Unexpected token"): {
+            return res.status(500).send({
+              success: false,
+              error: "An error occurred. Please try again.",
+            })
```

Comment:
Pulled all the relevant ones [from here](https://github.com/artsy/force/blob/damassi/fix/challenge-500/src/Components/AuthDialog/Utils/formatErrorMessage.ts)

Reply:
question (non-blocking): do you happen to know what caused `Unexpected token '<', "<!DOCTYPE "... is not valid JSON` and `Unexpected token '<', "<html> <"... is not valid JSON`?

Reply:
I _think_ it relates to the error page from api.artsy.net? At times error page html can come down the wire instead of JSON responses.